projects/gigya/src/cms-components/gigya-js/gigya-js.component.ts

```diff
@@ -0,0 +1,107 @@
+import { Component, OnInit, ChangeDetectorRef } from '@angular/core';
+import { BaseSiteService, LanguageService, ExternalJsFileLoader } from '@spartacus/core';
+import { GigyaConfig } from '../../config/gigya-config';
+
+declare var gigya: any;
+declare var window: Window;
+
+interface Window {
+  gigyaSpaLogin?: any;
+  GigyaRaasComponentReference?: any;
+  gigya?: any;
+  zone?: any;
+  __gigyaConf?: any;
+}
+
+window.__gigyaConf = { include: 'id_token' };
+
+window.gigyaSpaLogin = function (response) {
+  window.GigyaRaasComponentReference.zone.run(() =>
+    window.GigyaRaasComponentReference.component.onLoginEventHandler(response)
+  );
+};
+
+
+@Component({
+  selector: 'cx-gigya-js',
+  templateUrl: './gigya-js.component.html',
+})
+export class GigyaJsComponent implements OnInit {
```

Comment:
Instead of keeping this as a component I would change that to a service and initialize that on APP_INITIALIZER. I created example in: https://github.com/SAP/spartacus/pull/7824
Then we don't have to create custom Storefront component.

Reply:
Removed the GigyaJSComponent and moved all the functionality to GigyaJSService.